include/gauge_field.h

```diff
@@ -107,121 +122,120 @@ namespace quda {
 	    errorQuda("Error: invalid link type(%d)\n", link_type);
 	  }
 	}
-    
-    /**
-       @brief Helper function for setting the precision and corresponding
-       field order for QUDA internal fields.
-       @param precision The precision to use 
-     */
-    void setPrecision(QudaPrecision precision, bool force_native=false) {
-      // is the current status in native field order?
-      bool native = force_native ? true : false;
-      if (precision == QUDA_DOUBLE_PRECISION) {
-	if (order  == QUDA_FLOAT2_GAUGE_ORDER) native = true;
-      } else if (precision == QUDA_SINGLE_PRECISION ||
-		 precision == QUDA_HALF_PRECISION) {
-	if (reconstruct == QUDA_RECONSTRUCT_NO) {
-	  if (order == QUDA_FLOAT2_GAUGE_ORDER) native = true;
-	} else if (reconstruct == QUDA_RECONSTRUCT_12 || reconstruct == QUDA_RECONSTRUCT_13) {
-	  if (order == QUDA_FLOAT4_GAUGE_ORDER) native = true;
-	} else if (reconstruct == QUDA_RECONSTRUCT_8 || reconstruct == QUDA_RECONSTRUCT_9) {
-	  if (order == QUDA_FLOAT4_GAUGE_ORDER) native = true;
-	} else if (reconstruct == QUDA_RECONSTRUCT_10) {
-	  if (order == QUDA_FLOAT2_GAUGE_ORDER) native = true;
-	}
-      }
-
-      this->precision = precision;
-      this->ghost_precision = precision;
-
-      if (native) {
-	order = (precision == QUDA_DOUBLE_PRECISION ||
-                 reconstruct == QUDA_RECONSTRUCT_NO ||
-                 reconstruct == QUDA_RECONSTRUCT_10) ?
-	  QUDA_FLOAT2_GAUGE_ORDER : QUDA_FLOAT4_GAUGE_ORDER;
-      }
-    }
 
+        /**
+           @brief Helper function for setting the precision and corresponding
+           field order for QUDA internal fields.
+           @param precision The precision to use
+         */
+        void setPrecision(QudaPrecision precision, bool force_native = false)
+        {
+          // is the current status in native field order?
+          bool native = force_native ? true : false;
+          if (precision == QUDA_DOUBLE_PRECISION) {
+            if (order == QUDA_FLOAT2_GAUGE_ORDER) native = true;
+          } else if (precision == QUDA_SINGLE_PRECISION || precision == QUDA_HALF_PRECISION) {
```

Comment:
Should we add `QUDA_QUARTER_PRECISION` here too? Are there other templates we need to instantiate because of this change?

Reply:
Good catch Evan.  I'll add it now.  It won't have any other impact anywhere else.  Likely this was being done manually otherwise, though likely just wasn't noticed since quarter precision is used so sparingly and experimentally.